### PR TITLE
[RFC] Don't rely on :language

### LINKED ISSUE #187: I got some error message, when I use this plugin in win7+gvim.
I can not change my vim message language, so the error message is
qq 20160915190108

here is my debug step:
add `let g:wsdjeg = signlist` after line 27 of autoload/sy/sign.vim.
then `:echo g:wsdjeg` I got
qq 20160915190842

but in sign.vim
it is `let tokens = matchlist(signline, '\v^\s+line\=(\d+)\s+id\=(\d+)\s+name\=(.*)$')`
so I think maybe it is due to my language setting.

but I have this in my vimrc

``` vim
"Use English for anything in vim
if WINDOWS()
    silent exec 'lan mes en_US.UTF-8'
elseif OSX()
    silent exec 'language en_US'
else
    let s:uname = system("uname -s")
    if s:uname == "Darwin\n"
        " in mac-terminal
        silent exec 'language en_US'
    else
        " in linux-terminal
        silent exec 'language en_US.utf8'
    endif
endif

" try to set encoding to utf-8
if WINDOWS()
    " Be nice and check for multi_byte even if the config requires
    " multi_byte support most of the time
    if has('multi_byte')
        " Windows cmd.exe still uses cp850. If Windows ever moved to
        " Powershell as the primary terminal, this would be utf-8
        set termencoding=cp850
        " Let Vim use utf-8 internally, because many scripts require this
        set encoding=utf-8
        setglobal fileencoding=utf-8
        " Windows has traditionally used cp1252, so it's probably wise to
        " fallback into cp1252 instead of eg. iso-8859-15.
        " Newer Windows files might contain utf-8 or utf-16 LE so we might
        " want to try them first.
        set fileencodings=ucs-bom,utf-8,utf-16le,cp1252,iso-8859-15
    endif

else
    " set default encoding to utf-8
    set encoding=utf-8
    set termencoding=utf-8
endif
set langmenu=en_US
let $LANG = 'en_US'
source $VIMRUNTIME/delmenu.vim
source $VIMRUNTIME/menu.vim
```

### PR DESCRIPTION
**Pull request by mhinz**
Fixes #187.

### CODE REVIEW

File: autoload/sy/sign.vim [hunk 1]
```diff
@@ -24,15 +24,12 @@ function! sy#sign#get_current_signs() abort
...
-  let lang = v:lang
-  language message C
   redir => signlist
     silent! execute 'sign place buffer='. b:sy.buffer
   redir END
-  silent! execute 'language message' lang
 
   for signline in split(signlist, '\n')[2:]
-    let tokens = matchlist(signline, '\v^\s+line\=(\d+)\s+id\=(\d+)\s+name\=(.*)$')
+    let tokens = matchlist(signline, '\v^\s+\w+\=(\d+)\s+\w+\=(\d+)\s+\w+\=(.*)$')
```
Comment (jamessan) [inline autoload/sy/sign.vim]: Testing this on some of the po files, it doesn't work well on the various Asian languages I've tried.  The `\w+` should be changed to `\S+`.

Comment (jamessan) [inline autoload/sy/sign.vim]: Testing by using `:echo matchlist(getline('.'), '\v"\s+\S+\=(\%ld)\s+\S+\=(\%d)\s+\S+\=(.*)"')` on the translation lines (e.g., `msgstr "    行=%ld  識別子=%d  名前=%s"`).

Comment (wsdjeg) [inline autoload/sy/sign.vim]: yeah, here is my test script:
```viml
fu! Test()
    let str_en = '    line=12 id=2 name=SignifyChange'
    let str_cn = '    行=12 id=2 名称=SignifyChange'
    let tks_en = matchlist(str_en, '\v^\s+\S+\=(\d+)\s+\S+\=(\d+)\s+\S+\=(.*)$')
    let tks_cn = matchlist(str_cn, '\v^\s+\S+\=(\d+)\s+\S+\=(\d+)\s+\S+\=(.*)$')
    echom string(tks_en)
    echom string(tks_cn)
endf
```

Comment (wsdjeg) [inline autoload/sy/sign.vim]: here is result:
```
['    line=12 id=2 name=SignifyChange', '12', '2', 'SignifyChange', '', '', '', '', '', '']                                                            
['    行=12 id=2 名称=SignifyChange', '12', '2', 'SignifyChange', '', '', '', '', '', '']
```

Comment (mhinz) [inline autoload/sy/sign.vim]: Ah, thanks for the confirmation. I kind of suspected that `\w` wasn't working for e.g. Asian languages.

Now I tested this against: `  行=163 識別子=260 名前=SignifyAdd` now.

**Comment by mhinz** [comment]
Thanks!

**Pull request merged**